app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user 
+      @token = user.token
+    else
+      redirect_to expired_token_path 
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
```

Comment:
Its recommended to use `find_by` instead of `where(...).first` for single record search. Read the note at the end of [this assignment](https://www.gotealeaf.com/lessons/a3a6fbb9/assignments/1566).